ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/QueryFilterNode.java

```diff
@@ -358,6 +363,41 @@ public Void visitComparisonExpression(
       }
     }
 
+    @Override
+    public Void visitLikePredicate(final LikePredicate node, final Object context) {
+      final UnqualifiedColumnReferenceExp column = (UnqualifiedColumnReferenceExp) node.getValue();
+      if (column == null) {
+        setTableScanOrElseThrow(() -> invalidWhereClauseException("Like condition must directly "
+            + "reference a key column", isWindowed));
+      }
+      final ColumnName columnName = column.getColumnName();
+      final Column col = schema.findColumn(columnName)
+          .orElseThrow(() -> invalidWhereClauseException(
+              "Like condition on non-existent column " + columnName, isWindowed));
+
+      if (SqlBaseType.STRING != col.type().baseType()) {
+        throw invalidWhereClauseException("The column type for Like "
+            + "condition must be VARCHAR. The column type is "
+                + col.type().baseType().toString(), isWindowed);
+      }
+      return null;
+    }
+
+    @Override
+    public Void visitBetweenPredicate(final BetweenPredicate node, final Object context) {
+      final UnqualifiedColumnReferenceExp column = (UnqualifiedColumnReferenceExp) node.getValue();
+      if (column == null) {
+        setTableScanOrElseThrow(() -> invalidWhereClauseException(
+            "Between condition must directly reference " + "a key column", isWindowed));
+      }
+      final ColumnName columnName = column.getColumnName();
+      schema.findColumn(columnName)
+          .orElseThrow(() -> invalidWhereClauseException(
+              "Between condition on non-existent column " + columnName, isWindowed));
+
+      return null;
```

Comment:
Similar to like, I don't think we do type checking of the bounds in `ExpressionTypeManager`. The complicating thing is that between can take multiple types.  I think we need to either extend `ExpressionTypeManager` or rewrite the expression similar to the in predicate and make this equivalent to (COL <= A AND COL >= B).  In fact, if we did that, we would get the type checking automatically since it's done correct for comparisons.

Reply:
Rewrite the expression to comparison.